keras/layers/core.py

```diff
@@ -56,7 +56,7 @@ def __init__(self, mask_value=0., **kwargs):
         self.mask_value = mask_value
 
     def compute_mask(self, inputs, mask=None):
-        return K.any(K.not_equal(inputs, self.mask_value), axis=-1)
+        return K.cast(K.any(K.not_equal(inputs, self.mask_value), axis=-1), 'bool')
```

Comment:
[`K.any()`](https://github.com/fchollet/keras/blob/bd9214a5470ace245a6915d4e6d448401360e5cf/keras/backend/tensorflow_backend.py#L1140) and [`K.all()`](https://github.com/fchollet/keras/blob/bd9214a5470ace245a6915d4e6d448401360e5cf/keras/backend/tensorflow_backend.py#L1157) in the tf backend explicitly cast to `uint8`.  I'll change them there.